src/components/Reown/SendTransactionRequest.js

```diff
@@ -0,0 +1,491 @@
+  },
+  errorIcon: {
+    fontSize: 48,
+  },
+  feedbackModalIcon: {
+    width: 48,
+    height: 48,
+  },
+});
+
+export const SendTransactionRequest = ({ sendTransactionRequest, onAccept, onReject }) => {
+  const { dapp = {}, data = {} } = sendTransactionRequest || {};
+  const { tokens: registeredTokens, unregisteredTokens, reown } = useSelector((state) => ({
+    tokens: state.tokens,
+    unregisteredTokens: state.unregisteredTokens,
+    reown: state.reown
+  }));
+
+  // Combine registered and unregistered tokens
+  const knownTokens = { ...registeredTokens, ...unregisteredTokens };
+
+  const dispatch = useDispatch();
+  const navigation = useNavigation();
+
+  // Track tokens we've already requested to prevent duplicate requests
+  const requestedTokensRef = useRef(new Set());
+
+  // State for decline modal
+  const [showDeclineModal, setShowDeclineModal] = useState(false);
+
+  // Get transaction status from Redux
+  const sendTxStatus = reown.sendTransaction?.status || REOWN_SEND_TX_STATUS.READY;
+  const sendTxRetrying = reown.sendTransaction?.retrying || false;
+
+  // Show decline confirmation modal
+  const onDeclineTransaction = () => {
+    setShowDeclineModal(true);
+  };
+
+  // Use back button handler
+  const { navigateBack } = useBackButtonHandler(
+    onDeclineTransaction,
+    sendTxStatus === REOWN_SEND_TX_STATUS.SUCCESSFUL
+  );
+
+  // Request information for any unknown tokens
+  useEffect(() => {
+    // Only process if we have data and it hasn't been processed yet
+    if (data && data.inputs?.length > 0) {
+      // Collect all token UIDs from both inputs and outputs
+      const unknownTokensUid = [];
+
+      // Check tokens in inputs
+      const inputTokenIds = data.inputs?.map((input) => input.token).filter(Boolean) || [];
+      // Check tokens in outputs
+      const outputTokenIds = data.outputs?.map((output) => output.token).filter(Boolean) || [];
+
+      // Combine all token IDs
+      const allTokenIds = [...new Set([...inputTokenIds, ...outputTokenIds])];
+
+      // Find unknown tokens that we haven't requested yet
+      allTokenIds.forEach((uid) => {
+        if (uid !== DEFAULT_TOKEN.uid
+          && !(uid in registeredTokens)
+          && !(uid in unregisteredTokens)
+          && !requestedTokensRef.current.has(uid)
+        ) {
+          unknownTokensUid.push(uid);
+          requestedTokensRef.current.add(uid);
+        }
+      });
+
+      // Request download of unknown tokens
+      if (unknownTokensUid.length > 0) {
+        dispatch(unregisteredTokensDownloadRequest({ uids: unknownTokensUid }));
+      }
+    }
+  }, [data, registeredTokens, unregisteredTokens, dispatch]);
+
+  // Check if tokens are still loading
+  const isTokenDataLoading = () => unregisteredTokens.isLoading;
+
+  // Update the getTokenSymbol function to use knownTokens
+  const getTokenSymbol = (tokenId) => {
+    if (!tokenId) {
+      return constants.DEFAULT_NATIVE_TOKEN_CONFIG.symbol;
+    }
```

Comment:
I think we changed this in the desktop wallet PR. Now we don't have a default and you just check if it's the default native token UID

Reply:
Refactored to be exactly like the hathor wallet